src/docfx/lib/log/ErrorList.cs

```diff
@@ -18,12 +17,22 @@ internal class ErrorList : ErrorBuilder, IReadOnlyList<Error>
 
         public override bool HasError => _items.Any(item => item.Level == ErrorLevel.Error);
 
-        public override void Add(Error error) => _items.Add(error);
+        public override void Add(Error error)
```

Comment:
- Remove the `Count` and `this[int index]` method as read operations also needs `lock`.
- lock `HasError` method